stock_customer_deposits/tests/test_stock_customer_deposit.py

```diff
@@ -0,0 +1,388 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo.exceptions import ValidationError
+from odoo.tests import Form
+
+from odoo.addons.product.tests.common import ProductCommon
+from odoo.addons.sales_team.tests.common import SalesTeamCommon
+
+
+class TestStockCustomerDeposits(ProductCommon, SalesTeamCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.customer_location = cls.env.ref("stock.stock_location_customers")
+        cls.warehouse = cls.env["stock.warehouse"].create(
+            {
+                "name": "Test Warehouse",
+                "code": "TEST",
+                "use_customer_deposits": True,
+            }
+        )
+        cls.productA = cls.env["product.product"].create(
+            {
+                "name": "Product A",
+                "type": "product",
+            }
+        )
+        cls.productB = cls.env["product.product"].create(
+            {
+                "name": "Product B",
+                "type": "product",
+            }
+        )
+        cls.partner1 = cls.env["res.partner"].create({"name": "Test Partner"})
+        cls.partner2 = cls.env["res.partner"].create({"name": "Test Partner2"})
+
+    def test_warehouse_activate_customer_deposit(self):
+        """Create warehouse with use customer deposits, after deactivate and activate again."""
+        warehouse = self.env["stock.warehouse"].create(
+            {
+                "name": "Test Warehouse2",
+                "code": "TST",
+                "use_customer_deposits": True,
+            }
+        )
+        # Check type operation
+        self.assertTrue(warehouse.customer_deposit_type_id)
+        type_operation = warehouse.customer_deposit_type_id
+        self.assertRecordValues(
+            warehouse.customer_deposit_type_id,
+            [
+                {
+                    "name": "Customer Deposit",
+                    "code": "incoming",
+                    "use_create_lots": False,
+                    "use_existing_lots": True,
+                    "default_location_src_id": warehouse.lot_stock_id.id,
+                    "default_location_dest_id": self.customer_location.id,
```

Comment:
Is it normal to have incoming picking type with location dest as customer ?

Reply:
No it isn't normal @rousseldenis. 
In fact, I have just realised that it is not necessary for the desired behaviour.